### PR TITLE
Change to auto-assign Firefox related projects/vendors to FoxSec

### PR DESCRIPTION
**Pull request by caggle**
The aim of this PR is to auto-assign RRA requests that are triggered via CASA to FoxSec team, _only if_ the product line is "Firefox". Currently this is done manually by the people in the rotation. This could be a small improvement.

Marking as draft for now as I'd like to test it first.

### CODE REVIEW

**Comment by caggle** [comment]
Here is what this proposed change does:
- If a RRA/VA request comes from CASA via integration, it checks the "Product Line" field. If this contains "firefox", then the bug is assigned to FoxSec (Julien) in Bugzilla.
- If RRA/VA requested manually in Bugzilla (i.e. not from CASA), it checks the first comment (comment 0) in Bugzilla, and searches for certain keywords that are added to `config.yaml`. If found, then the bug is assigned to FoxSec (Julien). This is fuzzy and likely won't be 100% accurate, however the keywords can be fine-tuned for better matching logic if required, without having to change the logic.
- Bug assignee for FoxSec can also be changed easily in the config, if desired.

I was able to perform some testing manually with these changes, and it looks like it's working as I intended it.

@claudijd @tristanweir @moz-jvehent It would be great if all or a combination of you could review this change.

File: config.yaml [hunk 1]
```diff
@@ -19,3 +19,20 @@ casa:
     url: "https://biztera.com/api/v1/"
     lookup_period_in_days: 30
     bot_email: 'bizbot@biztera.com'
+foxsec:
+    assignee:
+        - culucenk@mozilla.com
```
Comment (claudijd) [inline config.yaml]: Should the auto-assign got to as relevant FoxSec triager rather than Cag?

File: config.yaml [hunk 2]
```diff
@@ -19,3 +19,20 @@ casa:
     url: "https://biztera.com/api/v1/"
     lookup_period_in_days: 30
     bot_email: 'bizbot@biztera.com'
+foxsec:
+    assignee:
```
Comment (claudijd) [inline config.yaml:23]: Should we make this "assignees" rather than a static assignment to a single individual?  We could have 1 person, but this precludes us from having many if we so choose.

File: assigner.py [hunk 3]
```diff
@@ -387,9 +403,9 @@ def main():
     logger.debug("Selected modules to run: {}".format(modules))
 
     if "rra" in modules:
-        autoassign(bapi, capi, config["bugzilla"]["rra"], args.dry_run)
+        autoassign(bapi, capi, config["bugzilla"]["rra"], config["casa"], config["foxsec"], args.dry_run)
     if "va" in modules:
-        autoassign(bapi, capi, config["bugzilla"]["va"], args.dry_run)
+        autoassign(bapi, capi, config["bugzilla"]["va"], config["casa"], config["foxsec"], args.dry_run)
```
Comment (claudijd) [inline assigner.py:403]: This may be a bit of scope creep, but I wonder if our param passing usage is outliving it's historical utility.  Would it be worth creating a small wrapper class that can read the config and allow us to simplify the param passing here?

Comment (claudijd) [reply assigner.py:403]: The argument is more or less that it would allow us to unit-test the config expectations more clearly and when/if we want the YAML schema to change it would honor the interface to what uses it.

File: config.yaml (hunk 1 above)
Comment (caggle) [reply config.yaml]: Pretty sure I changed this in a more recent commit to `jvehent@mozilla.com`. I just used my address for testing first.

File: config.yaml (hunk 2 above)
Comment (caggle) [reply config.yaml:23]: Happy to make the change to `assignees` to be consistent, however the code for this does not support picking from a list (using pickle, like the other list in the config file). I tried to keep it simple.

File: assigner.py (hunk 3 above)
Comment (caggle) [reply assigner.py:403]: I agree with you, but I am making this PR purely for "maintenance" purposes at this time. 

I think this entire program/script should be replaced/re-written using a more modern tech stack (likely serverless/lambdas) which would bring many benefits.

**Comment by caggle** [comment]
@moz-jvehent @claudijd thank you for the review! I replied and addressed the comments as applicable.
